service/frontend/adminHandler.go

```diff
@@ -445,10 +445,15 @@ func (adh *adminHandlerImpl) deleteWorkflowFromHistory(
 			logger.Error("Cannot decode thrift object", tag.Error(err))
 			continue
 		}
+		domainName, err := adh.GetDomainCache().GetDomainName(mutableState.ExecutionInfo.DomainID)
+		if err != nil {
+			return false
+		}
```

Comment:
this is probably incorrect / needs a log at the very least, otherwise we don't know it happened